### PR TITLE
Added rigid rotation model to `diff_rot()`

### LINKED ISSUE #5123: Add a rigid rotation model to the `diff_rot()`
I think it'd be useful to provide a rigid rotation model to `sunpy.physics.differential_rotation.diff_rot()`.  While, strictly speaking, it wouldn't have any differential rotation, it would allow a user to use the same machinery (e.g., `RotatedSunFrame`) with and without differential rotation to see what the differences are.  The sidereal rotation rate is available at `sunpy.sun.contants.sidereal_rotation_rate`.

### PR DESCRIPTION
**Pull request by asinghgaba**
### Description

Added a new parameter `"rigid"` to rot_params, which calculates longitude by assuming sun as a rigid body.
Fixes #5123

### CODE REVIEW

File: changelog/5132.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Added Rigid Rotation Model to :func:`~sunpy.physics.differential_rotation.diff_rot`, which can be used by passing `rot_type` as `"rigid"`.
```
Comment (nabobalis) [inline changelog/5132.feature.rst:1]:
```suggestion
Added Rigid Rotation Model to :func:`~sunpy.physics.differential_rotation.diff_rot`, which can be used by passing ``rot_type`` as "rigid".
```

File: sunpy/physics/differential_rotation.py [hunk 2]
```diff
@@ -115,10 +118,11 @@ def diff_rot(duration: u.s, latitude: u.deg, rot_type='howard', frame_time='side
 
     rot_params = {'howard': [2.894, -0.428, -0.370] * u.urad / u.second,
                   'snodgrass': [2.851, -0.343, -0.474] * u.urad / u.second,
-                  'allen': [14.44, -3.0, 0] * u.deg / u.day
+                  'allen': [14.44, -3.0, 0] * u.deg / u.day,
+                  'rigid': [sidereal_rotation_rate.value, 0, 0] * u.deg / u.day
```
Comment (ayshih) [inline sunpy/physics/differential_rotation.py:122]:
```suggestion
                  'rigid': sidereal_rotation_rate * [1, 0, 0]
```

File: changelog/5132.feature.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Added Rigid Rotation Model to :func:`~sunpy.physics.differential_rotation.diff_rot`, which can be used by passing ``rot_type`` as "rigid".
```
Comment (ayshih) [inline changelog/5132.feature.rst:1]:
```suggestion
Added a rigid rotation model to :func:`~sunpy.physics.differential_rotation.diff_rot` via ``rot_type=rigid``, where the rotation rate does not vary with latitude.
```

File: sunpy/physics/tests/test_differential_rotation.py [hunk 4]
```diff
@@ -130,6 +130,11 @@ def test_snodgrass(seconds_per_day):
     assert_quantity_allclose(rot, 135.4232 * u.deg, rtol=1e-3)
 
 
+def test_rigid(seconds_per_day):
+    rot = diff_rot(10 * seconds_per_day, 30 * u.deg, rot_type='rigid')
```
Comment (ayshih) [inline sunpy/physics/tests/test_differential_rotation.py:134]:
```suggestion
    rot = diff_rot(10 * seconds_per_day, [0, 30, 60] * u.deg, rot_type='rigid')
```
Might as well test multiple latitudes to confirm that the output doesn't vary with latitude

**Review by ayshih — APPROVED** [review]
Minor stuff, otherwise good to go

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks again @asinghgaba

**Comment by asinghgaba** [comment]
pleasure @nabobalis

**Pull request merged**